tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11135,402 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+    return std::stof(number);
+  } else if constexpr (std::is_same_v<T, double>) {
+    return std::stod(number);
+  }
+  else
+  {
+    VERIFY_IS_TRUE(false, L"Unsupported type for parsing string to number.");
+    return 0;
+  }
+}
+
+// Parse a vector of WEX::Common::Strings to a vector of numbers.
+// Expected string format is a comma separated list of numbers.
+// Strings with no commas could also be used for single numbers.
+// For example:
+// Three strings: ["1.0,2.0,3.0,4.0", "5.0,6.0,7.0,8.0", "9.0,10.0,11.0,12.0"]
+// would parse to 3 vectors of 4 elements each.
+// Or three strings ["1" , "2", "3"] will parse to 3 single elements vectors.
+// Vectors for a single element is a little weird, but it allows this helper
+// function to be very flexible.
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+std::vector<std::vector<T>> parseStringsToNumbers(const std::vector<WEX::Common::String>* input) {
+  std::vector<std::vector<T>> result = {};
+  if (input == nullptr || input->empty()) {
+      return result;
+  }
+
+  for (const auto& wexStr : *input) {
+
+    std::wstring wStr(wexStr);
+    std::wstringstream wss(wStr);
+    std::wstring number;
+    std::vector<T> vec;
+
+    while (std::getline(wss, number, L',')) {
+        vec.push_back(parseStringToNumber<T>(number));
+    }
+    result.push_back(vec);
+  }
+
+  // Sanity check for bad test data.
+  VERIFY_IS_TRUE(result.size() == input->size(), L"Result vector size does not match input vector size.");
+
+  return result;
+}
+
+TEST_F(ExecutionTest, LongVector_BinaryOpTest) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorBinaryOpTestBase<float, 4>();
+}
+
+template <typename T, std::size_t N>
+//typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+void ExecutionTest::LongVectorBinaryOpTestBase() {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+
+  // TODOLONGVEC: This seems frivolous as the device we target is created
+  // elsewhere. There is no guarantee that we're checking the right device here.
+  // Also, is there any helper code in this class already doing a similar check?
+  // This seems like something that should alreayd exist.
+  CComPtr<ID3D12Device> pDevice;
+  if (!m_ExperimentalModeEnabled  && !CreateDevice(&pDevice, D3D_SHADER_MODEL_6_9)) {
+    // TODOLONGVEC: This should fail. HLK requirements should select the test.
+    // But for exectests we can just return and skip. Will need to add a
+    // mechanism to faciliate that behavior in a good way that enforces it to
+    // always be on in OS builds.
+    WEX::Logging::Log::Comment("Device does not support SM 6.9. Can't run these tests.");
+    WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+    return;
+  }
+
+  // TableParameterHandler is a helper that parses the xml file linked
```

Comment:
I'm considering refactoring this into a helper function that parses the args into a struct based off of the applicable 'LongVector*Parameters' table definition for easier use amongst test cases.

Reply:
Went a different route with code cleanup.